### PR TITLE
Removed pose topic from log system

### PR DESCRIPTION
**Pull request by jennuine**
Signed-off-by: Jenn Nguyen <jenn@openrobotics.org>

# 🦟 Bug fix

Fixes https://github.com/ignitionrobotics/ign-gazebo/issues/753

## Summary
With changes from #742 , pose information was being logged twice (in the `dynamic_pose/info` and `changed_state` topics). This PR removes the recording `dynamic_pose/info` topic from `LogRecord` and updates the integration test.

~~`LogPlayback` still uses pose messages for backward compatibility (i.e., users have log files before #742 changes) but should probably be removed when ported to Fortress.~~

EDIT: PR also removes pose from `LogPlayback`

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/systems/log/LogPlayback.cc
Comment (j-rivero) [inline src/systems/log/LogPlayback.cc:627]: Could we introduce code that makes what the comments says automatically? I mean: check for major version, if it is 5 or greater skip the code.

**Review by j-rivero — COMMENTED** [review]
Good work of simplifying things.

> With changes from #742 , pose information was being logged twice (in the `dynamic_pose/info` and `changed_state` topics). This PR removes the recording `dynamic_pose/info` topic from `LogRecord` and updates the integration test.

Could we please get a Changelog and Migration entry for this effect?

Going to merge the PR with ign-gazebo4 to get some fresh CI.

File: src/systems/log/LogPlayback.cc
Comment (jennuine) [reply src/systems/log/LogPlayback.cc:627]: My plan was to immediately start forward porting this and removing the pose info for Fortress once this is merged. Should I still implement this?

Comment (j-rivero) [reply src/systems/log/LogPlayback.cc:627]: if that is the case, I think we don't need it.

**Comment by jennuine** [comment]
> Could we please get a Changelog and Migration entry for this effect?

How's this a3438b62b7f9643ce33ae96e1540b4eb02dfb82d?

**Comment by j-rivero** [comment]
> How's this [a3438b6](https://github.com/ignitionrobotics/ign-gazebo/commit/a3438b62b7f9643ce33ae96e1540b4eb02dfb82d)?

I think is fine, thanks @jennuine . If I understand correctly, anyone using the old topic should just change to `changed_state`. Is this correct?

File: test/integration/log_system.cc
Comment (j-rivero) [inline test/integration/log_system.cc:701]: Is this no longer true?

File: test/integration/log_system.cc
Comment (j-rivero) [inline test/integration/log_system.cc:785]: We are removing a good number of checks here. My poor understanding make me the question: why not to keep it and just use the other topic name? Are this checks no longer needed?

**Review by j-rivero — COMMENTED** [review]
Probably unrelated to this PR, but if I run `make INTEGRATION_log_system  `, the result is:

```
gn-gazebo/build on  jennuine/remove_pose_record ❯ ./bin/INTEGRATION_log_system 
Running main() from /home/jrivero/code/ignition/ign-gazebo/test/gtest/src/gtest_main.cc
[==========] Running 12 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 12 tests from LogSystemTest
[ RUN      ] LogSystemTest.LogPlaybackStatistics
[Msg] Loading default world.
...
[Dbg] [EntityComponentManager.cc:688] Using components of type [4981278897826323946] / [ign_gazebo_components.WorldSdf].
[Msg] Loaded level [3]
[Msg] No systems loaded from SDF, loading defaults
[Err] [ServerConfig.cc:890] Failed to copy installed config [/usr/local/share/ignition/ignition-gazebo4/playback_server.config] to default config [/home/jrivero/.ignition/gazebo/playback_server.config].(file /usr/local/share/ignition/ignition-gazebo4/playback_server.config doesn't exist)
[Err] [SystemLoader.cc:66] Failed to load system plugin [ignition-gazebo-log-system] : couldn't find shared library.
```

Is this a known problem?

**Comment by jennuine** [comment]
> If I understand correctly, anyone using the old topic should just change to changed_state. Is this correct?

Correct.

> Probably unrelated to this PR, but if I run make INTEGRATION_log_system , the result is ... Is this a known problem?

The failure is coming from `LogSystemTest.LogPlaybackStatistics` which this PR doesn't affect (does not change playback only changes record) and is the first time I've seen this error. Also, that test is the first test in `log_system.cc` which leads me to think maybe there's something different with your setup?

File: test/integration/log_system.cc
Comment (jennuine) [reply test/integration/log_system.cc:701]: This is no longer true. From my understanding, `dynamic_pose/info` was being published at [60Hz from `SceneBroadcaster`](https://github.com/ignitionrobotics/ign-gazebo/blob/657ca6dca0d7b1200c284c10a5776fb4f6aad3df/src/systems/scene_broadcaster/SceneBroadcaster.cc#L554-L555) & the number of expected poses varied based on `ign-transport`'s throttle mechanism, which was why they were counted.

Now, the `changed_state` topic is being published (almost) every time `LogRecord::PostUpdate` is called:
https://github.com/ignitionrobotics/ign-gazebo/blob/a3438b62b7f9643ce33ae96e1540b4eb02dfb82d/src/systems/log/LogRecord.cc#L702-L704

This test in particular (`LogSystemTest.RecordAndPlayback`) is using a double pendulum model, which always has new poses (i.e., states) for joints, so for 1000 iterations we are expecting 1000 state messages:
https://github.com/ignitionrobotics/ign-gazebo/blob/a3438b62b7f9643ce33ae96e1540b4eb02dfb82d/test/integration/log_system.cc#L840

File: test/integration/log_system.cc
Comment (jennuine) [reply test/integration/log_system.cc:785]: These checks are no longer needed and no longer valid. These checks were used for the extracted `msgs::Pose_V` from the `dynamic_pose/info` topic, but now since `LogRecord` is no longer recording the topic these checks are no longer valid. I've removed these checks and added new tests to check the pose from the `msgs::SerializedStateMap` instead (which comes from the `changed_state` topic)

**Comment by mjcarroll** [comment]
> Probably unrelated to this PR, but if I run make INTEGRATION_log_system , the result is ... Is this a known problem?

There doesn't seem to be any history of it in the ign-gazebo4 builds that I saw.

File: test/integration/log_system.cc
Comment (j-rivero) [reply test/integration/log_system.cc:701]: Thanks for the explanation, Jenn!. Sounds good to me.

File: test/integration/log_system.cc
Comment (j-rivero) [reply test/integration/log_system.cc:785]: ack! Thanks.

**Review by nkoenig — APPROVED** [review]

**Comment by jennuine** [comment]
Hold off on merging until get a confirm from @nkoenig

**Comment by nkoenig** [comment]
Please don't merge this until after July 8th.

**Comment by chapulina** [comment]
I'm not sure about changing the contents of the logs in a stable release, downstream users may be counting on this topic. Can this be targeted at Fortress?

**Comment by jennuine** [comment]
> Can this be targeted at Fortress?

I believe so, what do you think @nkoenig?

**Comment by jennuine** [comment]
Retargeting to Fortress and will be removing pose from `LogPlayback`

**Comment by jennuine** [comment]
The new recorded `state.tlog` file is from `rolling_shapes.sdf` but with only the sphere and ground plane models so that the log file would be relatively small.

**Comment by jennuine** [comment]
c7c943c7c9cb32097aa02e250d3f624b7b62b6ab Left a comment to not re-record `state.tlog` file with `-r` flag and to pause sim before terminating. Previously, I had done this and it resulted in strange pose behavior. The log file was 5.7s long and in the `LogSystemTest.LogControl` test, after seeking to 5s (close to the end but not exactly), `x = -74` then after rewinding and calling `server.Run` for 2 iterations `x = -75`. If it was 3 iterations then the pose would be the expected initial pose `x=0`. When I re-recorded the log without `-r`, started/unpaused the sim manually, and made sure the program was paused before terminating everything worked as expected.

**Comment by j-rivero** [comment]
All changes looks good to me. I've merged the latest state of the branch and restart the Windows build to get fresh CI. Seems to be a flaky test in github actions.

If Windows build ends nicely, consider this approved. Nice work Jenn.

**Comment by jennuine** [comment]
> If Windows build ends nicely, consider this approved

Windows didn't end nicely :( but looking through other merged PRs similar tests failed for windows as well (including `LogSystemTest`)

**Comment by j-rivero** [comment]
> Windows didn't end nicely :( but looking through other merged PRs similar tests failed for windows as well (including `LogSystemTest`)

Oh, "nicely" in the context of Windows meant "compilation is ok" since we did not fix the failing tests yet.

**Review by j-rivero — APPROVED** [review]

**Pull request merged**
